client/src/templates/Challenges/exam/show.tsx

```diff
@@ -362,28 +359,65 @@ class ShowExam extends Component<ShowExamProps, ShowExamState> {
   };
 
   runExam = () => {
-    // TODO: show loader
-    // TODO: fetch exam from server/database
-    const newExam = this.state.generatedExam.map(q => {
-      return { question: q.question, answer: null };
-    });
+    const {
+      data: {
+        challengeNode: {
+          challenge: { prerequisites }
+        }
+      },
+      completedChallenges,
+      createFlashMessage
+    } = this.props;
+
+    // check that they have all the prerequisites before they can start
+    let qualifiedForExam = true;
+    let incompleteChallenge = '';
+
+    for (let i = 0; i < prerequisites.length; i++) {
+      const prerequisiteChallenge = prerequisites[i];
+      const isPrerequisiteCompleted = completedChallenges.find(
+        (completedChallenge: CompletedChallenge) =>
+          prerequisiteChallenge.id === completedChallenge.id
+      );
+
+      if (!isPrerequisiteCompleted) {
+        qualifiedForExam = false;
+        incompleteChallenge = prerequisiteChallenge.title;
+        break;
+      }
+    }
```

Comment:
I changed it to an alert instead of a flash message:
![Screen Shot 2023-06-21 at 12 42 39 PM](https://github.com/freeCodeCamp/freeCodeCamp/assets/20648924/694021b8-1629-4f38-9194-4aa6f8552f63)
![Screen Shot 2023-06-21 at 12 43 17 PM](https://github.com/freeCodeCamp/freeCodeCamp/assets/20648924/228ca0f1-d127-428d-af03-1bb012a40982)

Reply:
Cool, that looks nicer.